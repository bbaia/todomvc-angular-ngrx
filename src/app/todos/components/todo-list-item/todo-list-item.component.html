<li
  *ngIf="todo as todo"
  [class.completed]="todo.completed"
  [class.editing]="editing"
>
  <div class="view">
    <input
      class="toggle"
      type="checkbox"
      [checked]="todo.completed"
      (click)="toggle.emit(todo.id)"
    />
    <label (dblclick)="activeEditMode()">
      {{ todo.text }}
    </label>
    <button class="destroy" (click)="delete.emit(todo.id)"></button>
  </div>
  <input
    class="edit"
    type="text"
    #textInput
    [hidden]="editing"
    [value]="todo.text"
    (keyup.enter)="updateText(todo.id, textInput.value)"
    (blur)="updateText(todo.id, textInput.value)"
  />
</li>
