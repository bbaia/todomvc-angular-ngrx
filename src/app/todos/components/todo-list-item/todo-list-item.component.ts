import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Todo } from './../../models';

@Component({
  selector: 'app-todo-list-item',
  templateUrl: './todo-list-item.component.html',
})
export class TodoListItemComponent {
  @Input() todo: Todo | null = null;
  @Output() toggle = new EventEmitter<number>();
  @Output() update = new EventEmitter<{ id: number; text: string }>();
  @Output() delete = new EventEmitter<number>();

  // 'local state is fine'
  editing = false;

  updateText(todoId: number, text: string): void {
    if (text && text.trim() !== this.todo?.text) {
      this.update.emit({ id: todoId, text: text.trim() });
    }
    this.editing = false;
  }

  activeEditMode(): void {
    this.editing = true;
  }
}
