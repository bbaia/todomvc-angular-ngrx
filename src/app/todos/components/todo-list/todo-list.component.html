<section id="main" class="main" *ngIf="todos as todos">
  <div class="toogle-view" *ngIf="todos.length > 0"></div>
  <ul id="todo-list" class="todo-list">
    <app-todo-list-item
      *ngFor="let todo of todos; trackBy: todosTrackByFn"
      [todo]="todo"
      (toggle)="toggle.emit($event)"
      (update)="update.emit($event)"
      (delete)="delete.emit($event)"
    ></app-todo-list-item>
  </ul>
</section>
