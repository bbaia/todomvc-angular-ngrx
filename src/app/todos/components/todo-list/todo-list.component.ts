import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Todo } from '../../models';

@Component({
  selector: 'app-todo-list',
  templateUrl: './todo-list.component.html',
})
export class TodoListComponent {
  @Input() todos: Todo[] | null = null;
  @Output() toggle = new EventEmitter<number>();
  @Output() update = new EventEmitter<{ id: number; text: string }>();
  @Output() delete = new EventEmitter<number>();

  todosTrackByFn(index: number, item: Todo): number {
    return item.id;
  }
}
