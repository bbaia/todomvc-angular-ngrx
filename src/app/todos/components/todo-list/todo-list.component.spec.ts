import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TodoListItemComponent } from '../todo-list-item/todo-list-item.component';
import { TodoListComponent } from './todo-list.component';

describe('TodoListComponent', () => {
  let component: TodoListComponent;
  let fixture: ComponentFixture<TodoListComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [TodoListComponent, TodoListItemComponent],
    });
    fixture = TestBed.createComponent(TodoListComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeDefined();
  });

  it('should render', () => {
    component.todos = [
      {
        id: 0.1,
        text: 'Learn Angular',
        creationDate: new Date(),
        completed: true,
      },
      {
        id: 0.2,
        text: 'Learn NgRx',
        creationDate: new Date(),
        completed: true,
      },
      {
        id: 0.3,
        text: 'Learn Jest',
        creationDate: new Date(),
        completed: true,
      },
    ];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
