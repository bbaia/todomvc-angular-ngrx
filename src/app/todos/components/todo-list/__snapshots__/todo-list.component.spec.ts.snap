// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListComponent should render 1`] = `
<app-todo-list
  __ngContext__={[Function LRootView]}
  delete={[Function EventEmitter_]}
  todos={[Function Array]}
  toggle={[Function EventEmitter_]}
  update={[Function EventEmitter_]}
>
  <section
    class="main"
    id="main"
  >
    <div
      class="toogle-view"
    />
    <ul
      class="todo-list"
      id="todo-list"
    >
      <app-todo-list-item>
        <li
          class="completed"
        >
          <div
            class="view"
          >
            <input
              class="toggle"
              type="checkbox"
            />
            <label>
               Learn Angular 
            </label>
            <button
              class="destroy"
            />
          </div>
          <input
            class="edit"
            type="text"
          />
        </li>
      </app-todo-list-item>
      <app-todo-list-item>
        <li
          class="completed"
        >
          <div
            class="view"
          >
            <input
              class="toggle"
              type="checkbox"
            />
            <label>
               Learn NgRx 
            </label>
            <button
              class="destroy"
            />
          </div>
          <input
            class="edit"
            type="text"
          />
        </li>
      </app-todo-list-item>
      <app-todo-list-item>
        <li
          class="completed"
        >
          <div
            class="view"
          >
            <input
              class="toggle"
              type="checkbox"
            />
            <label>
               Learn Jest 
            </label>
            <button
              class="destroy"
            />
          </div>
          <input
            class="edit"
            type="text"
          />
        </li>
      </app-todo-list-item>
    </ul>
  </section>
</app-todo-list>
`;
